pkg/chain/chain.go

```diff
@@ -15,13 +15,17 @@ import (
 // Handle represents a handle to an ethereum blockchain.
 type Handle interface {
 	// Address returns client's ethereum address.
-	Address() common.Address
+	Address() common.Address // TODO: use implementation-agnostic type
 	// StakeMonitor returns a stake monitor.
 	StakeMonitor() (chain.StakeMonitor, error)
 	// BalanceMonitor returns a balance monitor.
 	BalanceMonitor() (chain.BalanceMonitor, error)
 	// BlockCounter returns a block counter.
 	BlockCounter() chain.BlockCounter
+	// Signing returns a signer interface allowing to sign and verify messages
+	// using the chain implementation-specific mechanism as well as to
+	// convert between public key and address.
+	Signing() chain.Signing
```

Comment:
Maybe we should rename it to `Signer() chain.Signer`? We've got `StakeMonitor`, `BalanceMonitor`, `BlockCounter`, the returned interface is a Signer.

Reply:
Yeah, I think it should be `Signer()` but `keep-core` has `Signing()` and wanted to keep it consistent.

Reply:
We need to update it some day in both `keep-core` and `keep-ecdsa`. We could do it in https://github.com/keep-network/keep-core/pull/2242/ but now it's too late.